src/Interpreters/Aggregator.h

```diff
@@ -912,29 +913,60 @@ class Aggregator final
         bool compile_aggregate_expressions;
         size_t min_count_to_compile_aggregate_expression;
 
+        struct StatsCollectingParams
+        {
+            StatsCollectingParams();
```

Comment:
What's default contructor used for? I suppose we can provide more gurantees that we won't get inconsistent StatsCollectingParams, if we either:
* remove default ctor,
* use `key != 0` instead of `collect_hash_table_stats_during_aggregation` (it can't be `collect_hash_table_stats_during_aggregation = true` and `key = 0` at the same time, isn't it? In this case `collect_hash_table_stats_during_aggregation` may be reucntant)
* make fileds const to be sure that we construct only valid params
* Or even create `StatsCollectingParams` only when `collect_hash_table_stats_during_aggregation = true`, otherwise leave it nullptr

Reply:
default ctor is used in those calls where we don't pass any statistics collection related params.

> use key != 0 instead of collect_hash_table_stats_during_aggregation

done